optimum/habana/transformers/models/t5/modeling_t5.py

```diff
@@ -0,0 +1,216 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import habana_frameworks.torch.core as htcore
+import torch
+import torch.nn as nn
+
+# from transformers.activations import ACT2FN
+from transformers.models.t5.modeling_t5 import (
+    T5DenseActDense,
+    T5DenseGatedActDense,
+    T5LayerCrossAttention,
+    T5LayerFF,
+    T5LayerSelfAttention,
+    T5Stack,
+)
+
+
+def gaudi_T5Attention_forward(
+    self,
+    hidden_states,
+    mask=None,
+    key_value_states=None,
+    position_bias=None,
+    past_key_value=None,
+    layer_head_mask=None,
+    query_length=None,
+    use_cache=False,
+    output_attentions=False,
+):
+    """
+    Self-attention (if key_value_states is None) or attention over source sentence (provided by key_value_states).
```

Comment:
Can you replace this with a link to the original method and a one-line explanation about the change you made please? Similar to this for instance: https://github.com/huggingface/optimum-habana/blob/7fbe2362655e6681359b685690f2dc15b186755d/optimum/habana/transformers/models/gpt2/modeling_gpt2.py#L11